impl/src/test/java/org/ehcache/loaderwriter/writebehind/AbstractWriteBehindTestBase.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.loaderwriter.writebehind;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.exceptions.BulkCacheWritingException;
+import org.junit.Test;
+
+/**
+ * @author Abhilash
+ *
+ */
+public abstract class AbstractWriteBehindTestBase {
+  
+  protected CacheManager cacheManager ;
+  protected Cache<String, String> testCache;
+  protected final WriteBehindTestLoaderWriter<String, String> loaderWriter = new WriteBehindTestLoaderWriter<String, String>();
+  
+  @Test
+  public void testWrites(){
+    testCache.put("test1", "test1");
+    testCache.put("test2", "test2");
+    testCache.put("test3", "test3");
+    testCache.remove("test2");
+    try {
+      Thread.sleep(500);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+    assertThat(loaderWriter.getData().get("test1").getValue(), is("test1"));
+    assertThat(loaderWriter.getData().get("test2"), nullValue());
+  }
+  
+  @Test
+  public void testBulkWrites(){
+    for(int i=0 ; i<10; i++)
+      testCache.put("test"+i, "test"+i);
+    
+    Map<String, String> entries = new HashMap<String, String>(10);
+    Set<String> keys = new HashSet<String>(10);
+    for(int i=10 ; i<20; i++) {
+      entries.put("test"+i, "test"+i);
+      keys.add("test"+i);
+    }
+    
+    testCache.putAll(entries);
+    try {
+      Thread.sleep(500);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
e.printStackTrace() is one of the worst things you can do in a test. If you cannot handle the exception, let it flow out of the method to make the test framework report it. But in this case, there shouldn't be a Thread.sleep() so this comment is only for pedantry.